File: Server/src/main/java/jk_5/nailed/coremod/NailedFMLPlugin.java
package jk_5.nailed.coremod;

import cpw.mods.fml.relauncher.FMLLaunchHandler;
import cpw.mods.fml.relauncher.IFMLLoadingPlugin;
import cpw.mods.fml.relauncher.IFMLLoadingPlugin.MCVersion;
import cpw.mods.fml.relauncher.IFMLLoadingPlugin.Name;
import cpw.mods.fml.relauncher.IFMLLoadingPlugin.TransformerExclusions;
import jk_5.nailed.NailedLog;

import java.util.Map;

/**
 * No description given
 *
 * @author jk-5
 */
@Name("Nailed|Core")
@MCVersion("1.7.2")
@TransformerExclusions({"jk_5.nailed.coremod.transformers.", "jk_5.nailed.coremod.asm."})
@SuppressWarnings("unused")
public class NailedFMLPlugin implements IFMLLoadingPlugin {

    public static boolean obfuscated = false;

    @Override
    public String[] getASMTransformerClass() {
        if(FMLLaunchHandler.side().isServer()){
            return new String[]{
                    "jk_5.nailed.coremod.transformers.ClassHeirachyTransformer",
                    "jk_5.nailed.coremod.transformers.MinecraftServerTransformer",
                    //"jk_5.nailed.coremod.transformers.S21PacketChunkDataTransformer",
                    "jk_5.nailed.coremod.transformers.VanillaSupportTransformer",
                    "jk_5.nailed.coremod.transformers.WorldServerMultiTransformer",
                    "jk_5.nailed.coremod.transformers.NetHandlerPlayServerTransformer"
            };
        }
        return new String[0];
    }

    @Override
    public String getAccessTransformerClass(){
        return "jk_5.nailed.coremod.transformers.NailedAccessTransformer";
    }

    @Override
    public String getModContainerClass() {
        return null;
    }

    @Override
    public String getSetupClass() {
        return null;
    }

    @Override
    public void injectData(Map<String, Object> data) {
        obfuscated = (Boolean) data.get("runtimeDeobfuscationEnabled");
        NailedLog.info("Obfuscated: {}", obfuscated);
    }
}

File: Server/src/main/scala/jk_5/nailed/coremod/NailedFMLPlugin.scala
package jk_5.nailed.coremod

import cpw.mods.fml.relauncher.IFMLLoadingPlugin._
import cpw.mods.fml.relauncher.IFMLLoadingPlugin
import java.util
import jk_5.nailed.NailedLog

/**
 * No description given
 *
 * @author jk-5
 */
object NailedFMLPlugin {
  var obfuscated = false
}

@Name("Nailed|Core")
@MCVersion("1.7.2")
@TransformerExclusions(Array("jk_5.nailed.coremod.transformers.", "jk_5.nailed.coremod.asm.", "scala."))
@SuppressWarnings("unused")
class NailedFMLPlugin extends IFMLLoadingPlugin{

  override def getASMTransformerClass = Array(
    "jk_5.nailed.coremod.transformers.ClassHeirachyTransformer",
    "jk_5.nailed.coremod.transformers.MinecraftServerTransformer",
    //"jk_5.nailed.coremod.transformers.S21PacketChunkDataTransformer",
    "jk_5.nailed.coremod.transformers.VanillaSupportTransformer",
    "jk_5.nailed.coremod.transformers.WorldServerMultiTransformer",
    "jk_5.nailed.coremod.transformers.DimensionManagerTransformer",
    "jk_5.nailed.coremod.transformers.NetworkSystemTransformer",
    "jk_5.nailed.coremod.transformers.NetHandlerPlayServerTransformer"
  )

  override def getAccessTransformerClass = "jk_5.nailed.coremod.transformers.NailedAccessTransformer"
  override def getModContainerClass: String = null
  override def getSetupClass: String = null
  override def injectData(data: util.Map[String, AnyRef]){
    NailedFMLPlugin.obfuscated = data.get("runtimeDeobfuscationEnabled").asInstanceOf[java.lang.Boolean]
    NailedLog.info("Obfuscated: {}", NailedFMLPlugin.obfuscated)
  }
}
